packages/presentation/lib/src/pages/home_page/widgets/movie_grid_item.dart

```diff
@@ -1,33 +1,62 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:presentation/src/pages/home_page/bloc/home_bloc.dart';
+import 'package:presentation/src/pages/home_page/model/home_page_movie.dart';
 import 'package:presentation/src/pages/home_page/widgets/movie_short_info.dart';
 import 'package:presentation/src/pages/home_page/widgets/movie_stars.dart';
 import 'package:presentation/utils/images_container.dart';
 
 class MovieGridItem extends StatelessWidget {
-  const MovieGridItem({Key? key}) : super(key: key);
+  final HomePageMovie movie;
+  final HomeBloc bloc;
+  final int index;
+
+  const MovieGridItem(
+      {Key? key, required this.movie, required this.bloc, required this.index})
+      : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    return Container(
+    return GestureDetector(
+      onTap: () => bloc.onItemTap(index),
+      child: Container(
+        width: double.infinity,
+        height: double.infinity,
+        child: Column(
+          children: [
+            buildImage(),
+            MovieStars(
+              fullStarsCount: movie.stars,
+              isBigStar: false,
+            ),
+            MovieShortInfo(
+              movieGenres: movie.genre,
+              movieName: movie.title,
+              parentsGuide: movie.certification,
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Image buildImage() {
+    return Image.network(
+      movie.image,
       width: double.infinity,
-      height: double.infinity,
-      child: Column(
-        children: [
-          Image.asset(
-            AssetsImages.testImage,
-            width: double.infinity,
-            fit: BoxFit.fill,
-          ),
-          MovieStars(
-            fullStarsCount: 4,
-            isBigStar: false,
-          ),
-          MovieShortInfo(
-            movieGenres: 'Action',
-            movieName: 'John Wick',
-            parentsGuide: 'R',
-          ),
-        ],
+      height: 250,
+      fit: BoxFit.fitHeight,
+      loadingBuilder: (context, child, progress) {
+        return progress == null
+            ? child
+            : Center(child: const CircularProgressIndicator());
+      },
+      errorBuilder: (context, exception, stackTrace) => SvgPicture.asset(
+        AssetsImages.errorImage,
+        height: 250,
+        width: double.infinity,
+        color: Theme.of(context).colorScheme.onPrimary,
+        fit: BoxFit.fitHeight,
```

Comment:
put a value fit: BoxFit.Fill/Cover, remove properties "height" and use the "childspectratio" property in the "GridView.Builder" widget to adjust your movie card, this will give adaptability and get rid of 2 lines of code :)

Reply:
I use progress bar in my case) while image loading